src/server/models/cacheable_queries/question-response.js

```diff
@@ -68,11 +72,11 @@ const questionResponseCache = {
             .forUpdate()
             .where("question_response.campaign_contact_id", campaignContactId)
             .select("value", "interaction_step_id");
+          const newIds = {};
           const insertQuestionResponses = [];
           const updateStepIds = [];
-          const newIds = {};
           questionResponses.forEach(qr => {
-            newIds[qr.interactionStepId] = 1;
+            newIds[qr.interactionStepId.toString()] = 1;
```

Comment:
why are you casting to String in `newIds` but Number in toReturn?  I would vote do Number everywhere here, but let's be consistent one way or the other.